source/extensions/filters/http/jwt_authn/filter_config.cc

```diff
@@ -29,6 +33,32 @@ void FilterConfigImpl::init() {
           it.first, Verifier::create(it.second, proto_config_.providers(), *this));
     }
   }
+
+  std::vector<std::string> names;
+  for (const auto& it : proto_config_.requirement_map()) {
+    names.push_back(it.first);
+    name_verifiers_.emplace(it.first,
+                            Verifier::create(it.second, proto_config_.providers(), *this));
+  }
+  // sort is just for unit-test since protobuf map order is not deterministic.
+  std::sort(names.begin(), names.end());
+  all_requirement_names_ = absl::StrJoin(names, ",");
+}
+
+std::pair<const Verifier*, std::string>
+FilterConfigImpl::findPerRouteVerifier(const PerRouteFilterConfig& per_route) const {
+  if (per_route.config().bypass()) {
+    return std::make_pair(nullptr, EMPTY_STRING);
+  }
+
+  const auto& it = name_verifiers_.find(per_route.config().requirement_name());
+  if (it != name_verifiers_.end()) {
+    return std::make_pair(it->second.get(), EMPTY_STRING);
+  }
+
+  return std::make_pair(
+      nullptr, absl::StrCat("Wrong requirement_name: ", per_route.config().requirement_name(),
```

Comment:
In general this validation should happen when PerRouteConfig arrives, but not in per request path.

Reply:
Yes, but that is the difficulty.   In the FilterFactory,  CreateFilterFactory and CreatePerRouteObject are not tighten together. CreatePerRouteObject is called when RouteEntry is created,  CreateFilterFactory is called when your Filter is created.  They don't see each other.  Their config are suppose to be self-contain, one should not use the other.  But in order not to make too much duplication,  in this design,  `requirement_name` depends on the FilterConfig `requirement_map`.   The verification of `requirement_name` can not be done at config processing at main thread,   unfortunately, has to be done at worker thread, at request processing time.